nsxt/provider.go

```diff
@@ -158,13 +158,30 @@ func Provider() *schema.Provider {
 				Type:        schema.TypeString,
 				Optional:    true,
 				Description: "URL for VMC authorization service (CSP)",
-				DefaultFunc: schema.EnvDefaultFunc("NSXT_VMC_AUTH_HOST", "console.cloud.vmware.com/csp/gateway/am/api/auth/api-tokens/authorize"),
```

Comment:
I'm afraid we're not allowed to change defaults in order not to break backward compatibility. Is this change required?

Reply:
Technically the backward compatibility is not broken in that the same config (leaving out auth host in both config and env) will still work. The same default value is set in another place.

The reason I need to change this, is because oauth and access token need to use different CSP API URL to exchange for api token. The default value now needs to be adjusted based on auth method used (access token or oauth app). I checked that currently from schema this field is consumed in `getAPIToken`. I think it should be safe.